fuel-p2p/src/gossipsub/builder.rs

```diff
@@ -35,11 +33,12 @@ pub fn build_gossipsub(local_key: &Keypair, p2p_config: &P2PConfig) -> Gossipsub
         .mesh_n_high(p2p_config.max_mesh_size)
         .message_id_fn(gossip_message_id)
         .fast_message_id_fn(fast_gossip_message_id)
+        .validate_messages()
```

Comment:
Should we expose any settings to fine-tune the size and TTL of the message validation memcache?

Reply:
I'm thinking I expose `GossipsubConfig` struct from libp2p, that way we can set all the fields we want, instead of exposing only few ones and hiding the rest

Reply:
Instead of multiple gossipsub related fields in P2pConfig - I have exposed single `gossipsub_config: GossipsubConfig` field instead. I have also exposed `GossipsubConfigBuilder` to make it easy to set it up.